### PR TITLE
ci: merge staging to master

### PR DESCRIPTION
**Pull request by maxwell-aiden**
This is an automatic PR generated by the pipeline CI/CD. This will be automatically fast-forward merged if successful.

### CODE REVIEW

**Comment by maxwell-aiden** [comment]
Pipeline Attempt on 576559878 for 4a88ccb71d3c88ec0f41ff7e1dc66bca1286db6a

https://gitlab.com/MatrixAI/open-source/js-polykey/-/pipelines/576559878

**Comment by maxwell-aiden** [comment]
Pipeline Attempt on 576586049 for ef8dd1bed7a118473ec33f2b22711557c5bec797

https://gitlab.com/MatrixAI/open-source/js-polykey/-/pipelines/576586049

**Comment by maxwell-aiden** [comment]
Pipeline Attempt on 577531156 for 048e5d9bab12be5d0ebf429cf6f4bd291959ad8d

https://gitlab.com/MatrixAI/open-source/js-polykey/-/pipelines/577531156

**Comment by maxwell-aiden** [comment]
Pipeline Succeeded on 577531156 for 048e5d9bab12be5d0ebf429cf6f4bd291959ad8d

https://gitlab.com/MatrixAI/open-source/js-polykey/-/pipelines/577531156

File: tests/bin/utils.ts [hunk 1]
```diff
@@ -72,69 +73,97 @@ async function pkStdio(
...
-  // Parse the arguments of process.stdout.write and process.stderr.write
-  const parseArgs = (args) => {
-    const data = args[0];
-    if (typeof data === 'string') {
-      return data;
-    } else {
-      let encoding: BufferEncoding = 'utf8';
-      if (typeof args[1] === 'string') {
-        encoding = args[1] as BufferEncoding;
+  if (global.testCmd != null) {
```
Comment (CMCDragonkai) [inline tests/bin/utils.ts:76]: This should be `testCommand` to match the `PK_TEST_COMMAND` environment variable.

File: tests/bin/utils.ts [hunk 2]
```diff
@@ -72,69 +73,97 @@ async function pkStdio(
...
+      ...process.env,
+      ...env,
+    };
+    const command = path.resolve(path.join(global.projectDir, global.testCmd));
+    const subprocess = child_process.spawn(command, [...args], {
+      env,
+      cwd,
+      stdio: ['pipe', 'pipe', 'pipe'],
+      windowsHide: true,
+    });
```
Comment (CMCDragonkai) [inline tests/bin/utils.ts:88]: This now swaps `pkStdio` to use `spawn` in the case of targeting an external executable. This requires comments up top about how the `pkStdio` changes behaviour.

**Comment by CMCDragonkai** [comment]
@tegefaulkes disable `integration:merge` when working on staging CI.

**Pull request merged**